### PR TITLE
Onnx refactor: adding open_clip onnx support, decouple image preprocess

### PR DESCRIPTION
**Pull request by wanliAlex**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
features

* **What is the current behavior?** (You can also link to an open issue here)
we do not support open_clip onnx
we do not take the image preprocess out

* **What is the new behavior (if this is a feature change)?**
we now support open_clip onnx
the image preprocess is decoupled from the clip model

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
NO

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
will add soon

* **Related Python client changes** (link commit/PR here)
no

* **Related documentation changes** (link commit/PR here)
will add soon

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -15,7 +15,7 @@ Pillow==9.3.0
 numpy==1.23.4
 validators==0.20.0
 sentence-transformers==2.2.2
-open_clip_torch==2.0.2
+open_clip_torch==2.9.1
 clip-marqo==1.0.2
```
Comment (pandu-k) [inline requirements.txt:19]: Make sure to run the entire test suite to make sure this doesn't break anything

File: src/marqo/s2_inference/onnx_clip_utils.py [hunk 2]
```diff
@@ -13,60 +13,65 @@
...
 # Loading shared functions from clip_utils.py. This part should be decoupled from models in the future
-from marqo.s2_inference.clip_utils import get_allowed_image_types, format_and_load_CLIP_image, format_and_load_CLIP_images, load_image_from_path,_is_image
+from marqo.s2_inference.clip_utils import get_allowed_image_types, format_and_load_CLIP_image, \
+    format_and_load_CLIP_images, load_image_from_path, _is_image
 
 logger = get_logger(__name__)
 
-_HF_MODEL_DOWNLOAD = {
+OPENAI_DATASET_MEAN = (0.48145466, 0.4578275, 0.40821073)
+OPENAI_DATASET_STD = (0.26862954, 0.26130258, 0.27577711)
```
Comment (pandu-k) [inline src/marqo/s2_inference/onnx_clip_utils.py:30]: Can you add a comment about this mean/ std: where it came from and what it is for?

**Comment by jn2clark** [comment]
Thanks @wanliAlex ! My only suggestion is to have an error message if the model download does not work and have that passed through to the user

**Comment by wanliAlex** [comment]
> Thanks @wanliAlex ! My only suggestion is to have an error message if the model download does not work and have that passed through to the user

```python
try:
    file_path = hf_hub_download(repo_id=repo_id, filename=filename,
                                cache_dir=cache_dir)
except (EnvironmentError, OSError, ValueError, RepositoryNotFoundError, RevisionNotFoundError, EntryNotFoundError,
    LocalEntryNotFoundError) as e:
    raise ModelDownloadError(e)
```

error catch added

**Review by pandu-k — APPROVED** [review]

**Pull request merged**